libs/utils/src/file_reading.ts

```diff
@@ -0,0 +1,73 @@
+import JsZip, { JSZipObject } from 'jszip';
+import { Buffer } from 'buffer';
+
+export function readFileAsyncAsString(file: File): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      const { result } = reader;
+      resolve(typeof result === 'string' ? result : '');
+    };
+    reader.onerror = reject;
+    reader.readAsText(file);
+  });
+}
+
+export function readFile(file: File): Promise<Buffer> {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+
+    /* istanbul ignore next */
+    reader.onerror = () => {
+      reject(reader.error);
+    };
+
+    reader.onload = () => {
+      resolve(Buffer.from(reader.result as ArrayBuffer));
```

Comment:
I know the TS types say that `reader.result` can be any of `string | ArrayBuffer | null`. I imagine `string` isn't actually possible here since we're calling `reader.readAsArrayBuffer`, but `null` might be?

`Buffer.from(null)` will throw an error so might want to explicitly do `Buffer.from([])` (an empty Buffer) when the `reader.result` is `null`